src/Common/src/Interop/Unix/libssl/Interop.libssl_types.cs

```diff
@@ -61,12 +62,43 @@ internal static class SslSizes
             internal const int TRAILER_SIZE = 68;
         }
 
-        internal static class SslErrorCode
+        internal static class SslMethods
         {
-            internal const int SSL_ERROR_SSL = 1;
-            internal const int SSL_ERROR_WANT_READ = 2;
-            internal const int SSL_ERROR_SYSCALL = 5;
-            internal const int SSL_ERROR_ZERO_RETURN = 6;
+            internal static readonly IntPtr TLSv1_method;
+            internal static readonly IntPtr TLSv1_1_method;
+            internal static readonly IntPtr TLSv1_2_method;
+            internal static readonly IntPtr SSLv3_method;
+            internal static readonly IntPtr SSLv23_method;
+
+            private static IntPtr checkMethodPtr(IntPtr methodPtr)
```

Comment:
Actually, instead, I'd suggest doing this:
```C#
internal static readonly IntPtr TLSv1_method = libssl.TLSv1_method();
internal static readonly IntPtr TLSv1_1_method = libssl.TLSv1_1_method();
internal static readonly IntPtr TLSv1_2_method = libssl.TLSv1_2_method();
internal static readonly IntPtr SSLv3_method = libssl.SSLv3_method();
internal static readonly IntPtr SSLv23_method = libssl.SSLv23_method();

#if DEBUG
static SslMethods()
{
    Debug.Assert(TLSv1_method != IntPtr.Zero, "TLSv1_method is null");
    Debug.Assert(TLSv1_1_method != IntPtr.Zero, "TLSv1_1_method is null");
    Debug.Assert(TLSv1_2_method != IntPtr.Zero, "TLSv1_2_method is null");
    Debug.Assert(SSLv3_method != IntPtr.Zero, "SSLv3_method is null");
    Debug.Assert(SSLv23_method != IntPtr.Zero, "SSLv23 method is null");
}
#endif
```
That way, in release builds, SslMethods will be a beforefieldinit type, we won't end up paying anything for checkMethodPtr in release builds, and the assert will tell us in debug builds exactly which method is null.